### PR TITLE
pop-gtk-theme: init at 2019-11-13

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Simply includes a  [popular theme developed by System76](https://github.com/pop-os/gtk-theme). 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

(On recommendation of @worldofpeace on [this discourse thread](https://discourse.nixos.org/t/searching-for-package-submitting-guidelines/5066/3)) :

cc @worldofpeace @jtojnar @romildo @c0bw3b

### CODE REVIEW

File: pkgs/data/themes/pop-gtk/default.nix
Comment (pbogdan) [inline pkgs/data/themes/pop-gtk/default.nix]: You could simplify this a bit with a loop like so:

```
    for file in $(find -name render-asset\*.sh); do
      substituteInPlace "$file" \
        --replace 'INKSCAPE="/usr/bin/inkscape"' \
                  'INKSCAPE="${inkscape}/bin/inkscape"' \
        --replace 'OPTIPNG="/usr/bin/optipng"' \
                  'OPTIPNG="${optipng}/bin/optipng"'
    done
```

File: pkgs/data/themes/pop-gtk/default.nix
Comment (pbogdan) [inline pkgs/data/themes/pop-gtk/default.nix]: I believe this shouldn't be needed as the default builder will invoke make with an appropriate `SHELL`.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (pbogdan) [inline pkgs/data/themes/pop-gtk/default.nix]: Personally I've used

```
  installFlags = [
    "DESTDIR=${placeholder "out"}"
    "BASE_DIR=/share/themes"
  ];
```
Re-checking now I see there is an additional hardcoded `/usr/share/themes` path in the `install-gnome-shell` target but it doesn't get invoked by default.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (pbogdan) [inline pkgs/data/themes/pop-gtk/default.nix]: Appears to be unused.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (worldofpeace) [reply pkgs/data/themes/pop-gtk/default.nix]: Could we PR upstream just not using `/usr/bin/` at all?
They could just expect it to be in PATH, it makes more sense. Though it could be better even.

**Comment by pbogdan** [comment]
Added a few (hopefully helpful) suggestions based on the package I've had kicking around in a local overlay though probably best to defer to folks you CC'ed.  
Would personally welcome the inclusion of the package though so thank you for submitting the PR.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (c0bw3b) [inline pkgs/data/themes/pop-gtk/default.nix]: GitHub archives are sometimes regenerated over time and the hash changes over time, so you shouldn't use `fetchurl` here. Instead, use `fetchFromGitHub` or `fetchzip`.
The hash for those two fetchers needs to be computed over the content of the archive _after_ unpacking. You can precompute it with `nix-prefetch-url --type sha256 unpack <archive_url>`

File: pkgs/data/themes/pop-gtk/default.nix
Comment (c0bw3b) [inline pkgs/data/themes/pop-gtk/default.nix]:
```suggestion
    homepage    = "https://github.com/pop-os/gtk-theme";
```

We had a [recent decision](https://github.com/NixOS/rfcs/pull/45) to favor quoting URLs. It's not yet generalized in nixpkgs but since this is an init of a new pkg, let's do it right from the start. :)

File: pkgs/top-level/all-packages.nix
Comment (c0bw3b) [inline pkgs/top-level/all-packages.nix]: This file is already a bit messy, but further down there is `### DATA` section that contains the packages located under pkgs/data, such as themes.
So this line should rather be between `polytopes_db` and `posix_man_pages`

**Review by c0bw3b — COMMENTED** [review]
Left a few comments. :)

Your addition to the maintainers list should be a separate commit (and it needs to be before the pkg addition) with the message:
`maintainers: add elyhaka`

**Review by romildo — CHANGES_REQUESTED** [review]

File: pkgs/data/themes/pop-gtk/default.nix
Comment (romildo) [reply pkgs/data/themes/pop-gtk/default.nix]: This release is more than one year old. Upstream current development changed to the meson build system.

Comment (romildo) [reply pkgs/data/themes/pop-gtk/default.nix]: This release is more than one year old. Recently upstream development seems to have been changed to the meson build system.

Comment (worldofpeace) [reply pkgs/data/themes/pop-gtk/default.nix]: And, it's possible the theme is completely broken with gnome 3.34?
If I understand it correctly, that release probably targeted whatever was in ubuntu bionic.
So we might have to use the HEAD of `master_eoan` branch.

Comment (jtojnar) [reply pkgs/data/themes/pop-gtk/default.nix]: Uugh, they made the Makefile parametric in 4.0b2 but the meson port uses hardcoded paths in the scripts again.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Added it, thanks

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: > And it's possible the theme is completely broken with gnome 3.34?
> If I understand it correctly, that release probably targeted whatever was in ubuntu bionic.
> So we might have to use the HEAD of `master_eoan` branch.

I have started to work on supporting the v4, to anticipate the release but I submitted the 3.1.2 because it was the latest stable. Do you want me to submit the v4 directly even if it's a beta ?

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: I didn't knew file could changes, I switched to `fetchFromGitHub`, thanks !

File: pkgs/data/themes/pop-gtk/default.nix
Comment (c0bw3b) [inline pkgs/data/themes/pop-gtk/default.nix]: You can use `rev = version` here to fetch the archive from upstream `3.1.2` tagged release.

Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Fixed it !

File: pkgs/data/themes/pop-gtk/default.nix
Comment (romildo) [reply pkgs/data/themes/pop-gtk/default.nix]: It looks like that System76-Pop! OS distributes [a version based on HEAD](https://launchpad.net/~system76/+archive/ubuntu/pop/+sourcepub/10732034/+listing-archive-extra) of `master_eon`. Probably we should do the same.

Comment (worldofpeace) [reply pkgs/data/themes/pop-gtk/default.nix]: Sounds good @romildo as it looks we agreed. @Elyhaka Could you do this?

Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Of course, working on it !

Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Done, I have been able to remove some dependencies in the meantime.

Comment (worldofpeace) [reply pkgs/data/themes/pop-gtk/default.nix]: > Of course, working on it !

Oh sorry :smile:, I just wanted to ask properly because we have tendencies to just **suggest** instead stating "hey, this?

File: pkgs/data/themes/pop-gtk/default.nix
Comment (romildo) [inline pkgs/data/themes/pop-gtk/default.nix]: Usually we use the date of the commit as the version: 2019-11-13

**Review by romildo — CHANGES_REQUESTED** [review]

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Oh, I see ! I didn't took it like that, no problem :)

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Noted, thank you

**Review by romildo — APPROVED** [review]

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: Install flags are not used by meson

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: Why does it need the extra themes?

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: Is this still necessary with meson?

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: It is better to patch specific files, so that we underpatch rather than overpatch. The former will break the build, while the latter will fail silently.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: Wouldn't just the following be enough? 
```suggestion
                  'INKSCAPE="inkscape"' \
```

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: Meson already builds in parallel.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: Can you use `with licenses`?

File: pkgs/data/themes/pop-gtk/default.nix
Comment (jtojnar) [inline pkgs/data/themes/pop-gtk/default.nix]: `postPatch` is more customary place for changing sources.

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Right, removed it.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: No, it was to silence an error in inkscape when I was working on 3.1.2 version.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Not needed anymore, when working on 3.1.2 version it was required to have some elements displaying correctly (probably related to the fact that the latest stable version was completly out of date).

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Noted, I moved it in the loop for patching `render-*.sh` files.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: It works, indeed.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (worldofpeace) [inline pkgs/data/themes/pop-gtk/default.nix]:
```suggestion
      patchShebangs "$file"
```

rather important in this case :smile:

File: pkgs/data/themes/pop-gtk/default.nix
Comment (worldofpeace) [inline pkgs/data/themes/pop-gtk/default.nix]: I think there's an entry in the code style conventions that suggest alignment, but it's not really encouraged and editors can just render text that way without edits.

**Comment by worldofpeace** [comment]
Here's screenshots of the gtk-widget-factory with the theme on nixos-unstable:
Screenshot from 2019-12-12 05 30 59

Screenshot from 2019-12-12 05 31 16


Haven't checked out the gnome-shell theme or sound theme.

File: pkgs/data/themes/pop-gtk/default.nix
Comment (Elyhaka) [reply pkgs/data/themes/pop-gtk/default.nix]: Indeed, fixed it

**Comment by Elyhaka** [comment]
Pushed update with latest version

**Pull request merged**
